editor for tig

vbauerster opened the pull request:
It turns out that tig doesn't respect `$EDITOR` env var, at least I wasn't able to invoke kak with its `e` command within connect-terminal. Exporting `GIT_EDITOR` makes tig to invoke internal edit command.

vbauerster commented:
[Related](https://discuss.kakoune.com/t/rfr-roll-back-through-old-versions-of-a-file-in-git/743).

alexherbo2 commented:
Can you add a copyright waiver before the first commit and add this variable in `overrides` instead?

It could eventually go in a `git.env`, but `overrides` looks reasonable enough.  What do you think?

vbauerster commented:
Ok, I actually put it first in overrides, but it needs `export` so I decided to separate.
`GIT_EDITOR` env var is specific to [tig](https://jonas.github.io/tig/doc/tigrc.5.html) which tells tig to ignore `git core.editor` setting, hence the branch name. However I can rename if you propose a different name.

vbauerster committed:
Vladimir Bauer Copyright Waiver

I dedicate any and all copyright interest in this software to the
public domain. I make this dedication for the benefit of the public at
large and to the detriment of my heirs and successors. I intend this
dedication to be an overt act of relinquishment in perpetuity of all
present and future rights to this software under copyright law.

vbauerster committed:
Export GIT_EDITOR so tig is aware of internal edit command.

alexherbo2 commented:
`GIT_EDITOR` is a [Git environment variable][Git Internals – Environment Variables].

Not sure if it’s possible in the same PR to rename the branch.

[Git Internals – Environment Variables]: https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables

vbauerster commented:
I've made a change.
